src/OrleansAzureUtils/ISiloHostBuilderExtensions.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using Microsoft.Extensions.Configuration;
+using Orleans.AzureUtils.Configuration;
+using Orleans.Hosting;
+
+namespace OrleansAzureUtils
+{
+    public static class ISiloHostBuilderExtensions
+    {
+
+        /// <summary>
+        /// Configure DI to use AzureBasedMemebershipTable
+        /// </summary>
+        /// <param name="builder"></param>
+        /// <param name="configureOptions"></param>
+        public static ISiloHostBuilder UseAzureMemebershipTable(this ISiloHostBuilder builder,
```

Comment:
to be honest, I would drop the suffix Table from Membership. The end user doesn't care if this is implemented via a membership table, or via a different approach, such as in ServiceFabric.
And in particular for this implementation, I would be explicit that this membership uses AzureTable. So the name would end up being `UseAzureTableMembership()`. Do not confuse `Table` as being moved around, it's just that this implementation of membership uses Azure Table, and not Azure Blob nor CosmosDb, which the latter we'll likely implement at some point, so just Azure as the name would be ambiguous; AzureTable is not

Reply:
sounds good to me